### PR TITLE
Some changes in softlayer-go

### PR DESCRIPTION
**Pull request by jianqiu**
In this PR, in order to keep consistent with the softlayer api data types, made the following changes in `data_type/softlayer_product_order.go` file
1. refactored the name `softlayer_product_order.go` to `softlayer_container_product_order`
2. Add two types of softlayer_container_product_order 
    `SoftLayer_Container_Product_Order_Network_PerformanceStorage_Iscsi`
(http://sldn.softlayer.com/reference/datatypes/SoftLayer_Container_Product_Order_Network_PerformanceStorage_Iscsi)
   `SoftLayer_Container_Product_Order_Virtual_Guest_Upgrade`
(http://sldn.softlayer.com/reference/datatypes/SoftLayer_Container_Product_Order_Virtual_Guest_Upgrade)
3. Two more types of ordering in `services/softLayer_Product_Order_Service.go`
    `PlaceContainerOrderNetworkPerformanceStorageIscsi`
    `PlaceContainerOrderVirtualGuestUpgrade`
4. A new softlayer service `service/softlayer_network_storage_allowed_host_service.go`
(http://sldn.softlayer.com/reference/services/SoftLayer_Network_Storage_Allowed_Host)
5. A new method `GetAllowedHost` implemented in `service/softlayer_virtual_guest_service.go` 
(http://sldn.softlayer.com/reference/services/SoftLayer_Virtual_Guest/getAllowedHost)

### CODE REVIEW

**Comment by jianqiu** [comment]
@maximilien 

Could you please review this changes ? Thanks!

File: services/softlayer_network_storage.go [hunk 1]
```diff
@@ -45,10 +45,10 @@ func (slns *softLayer_Network_Storage_Service) CreateIscsiVolume(size int, locat
 
 	iopsItemPriceId := slns.getPerformanceStorageItemPriceIdByIops(size)
 
-	order := datatypes.SoftLayer_Product_Order{
+	order := datatypes.SoftLayer_Container_Product_Order_Network_PerformanceStorage_Iscsi{
 		Location:    location,
 		ComplexType: "SoftLayer_Container_Product_Order_Network_PerformanceStorage_Iscsi",
-		OsFormatType: datatypes.OsFormatType{
+		OsFormatType: datatypes.SoftLayer_Network_Storage_Iscsi_OS_Type{
 			Id:      12,
```
Comment (maximilien) [inline services/softlayer_network_storage.go]: I am assuming that `12` and `LINUX` are the defaults?

File: services/softlayer_network_storage.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func (slns *softLayer_Network_Storage_Service) DeleteIscsiVolume(volumeId int, i
 }
 
 func (slns *softLayer_Network_Storage_Service) GetIscsiVolume(volumeId int) (datatypes.SoftLayer_Network_Storage, error) {
-	response, err := slns.client.DoRawHttpRequest(fmt.Sprintf("%s/%d/getObject.json", slns.GetName(), volumeId), "GET", new(bytes.Buffer))
+	response, err := slns.client.DoRawHttpRequestWithObjectMask(fmt.Sprintf("%s/%d/getObject.json", slns.GetName(), volumeId), []string{"serviceResourceBackendIpAddress", "lunId"}, "GET", new(bytes.Buffer))
```
Comment (maximilien) [inline services/softlayer_network_storage.go]: Can you please explain this: `[]string{"serviceResourceBackendIpAddress", "lunId"},`

I think this is the first SL service where we have to pass such info?

**Comment by maximilien** [comment]
Hi, @jianqiu,

Looks really good. I only have a couple of questions...  Mostly so I can understand and make sure you double check that the values you are hardcoding are correct and should be hardcoded vs being parameters.

Once you acknowledge and answer, I will merge and update the Godeps for `bosh-softlayer-cpi` and `bosh-softlayer-stemcells`.

Best,

max

File: services/softlayer_network_storage.go (hunk 1 above)
Comment (jianqiu) [inline services/softlayer_network_storage.go]: @maximilien 
Here is a sample instance for `SoftLayer_Network_Storage_Iscsi_OS_Type `

`osType":{"createDate":"2014-07-16T21:22:25+08:00","description":"Use if your host operating system is Linux.","id":12,"keyName":"LINUX","name":"Linux"}`

The `Id` for `Linux` os type is `12`, that is really what we want during ordering the iscsi storage, and it is constant, no longer change in future, so I just hardcoded it here.

File: services/softlayer_network_storage.go (hunk 2 above)
Comment (jianqiu) [inline services/softlayer_network_storage.go]: @maximilien 

The first SL service without object mask will not retrieve `serviceResourceBackendIpAddress `,`lunId ` as defaults. Here is a sample result:
`{"accountId":278444,"capacityGb":500,"createDate":"2015-06-20T21:20:19+08:00","guestId":null,"hardwareId":null,"hostId":null,"id":5453539,"nasType":"ISCSI","serviceProviderId":1,"storageTypeId":"5","upgradableFlag":true,"username":"IBM01SL278444-336","serviceResourceName":"PerfStor Aggr aggr_staassjc0101_hp01"}`

If we want to fetch information including `serviceResourceBackendIpAddress ` and `lunId `, we need to call SL API with object mask.

`{"accountId":278444,"capacityGb":500,"createDate":"2015-06-20T21:20:19+08:00","guestId":null,"hardwareId":null,"hostId":null,"id":5453539,"nasType":"ISCSI","serviceProviderId":1,"storageTypeId":"5","upgradableFlag":true,"username":"IBM01SL278444-336","lunId":"101","serviceResourceBackendIpAddress":"10.1.206.78","serviceResourceName":"PerfStor Aggr aggr_staassjc0101_hp01"}`

**Comment by jianqiu** [comment]
Hi, @maximilien 

Could you refer to my comments, and review these changes again, thanks!

**Comment by maximilien** [comment]
Ok merging. I will update Godeps later tonight or tomorrow AM when I have access to my laptop.

Cheers :beers:

**Pull request merged**